### PR TITLE
Restructure playbooks, add tags, and add asset and db deploy, varnish now optional

### PR DESCRIPTION
**Pull request by nexpoehler**
I restructured environment.yml to add tags and use includes so it currently looks like this:

    [root@ansible ansible-playbook-cloud-app]# cat playbooks/environment.yml
    ---
    
    - include: _base.yml tags=base
    - include: _fs.yml tags=fs
    - include: _web.yml tags=web
    - include: _fpm.yml tags=fpm
    - include: _db.yml tags=db
    - include: _lb-web.yml tags=lb-web
    - include: _lb-fpm.yml tags=lb-fpm
    - include: _redis.yml tags=redis
    - include: _lb-redis.yml tags=lb-redis
    - include: _varnish.yml tags=varnish
    - include: _lb-varnish.yml tags=lb-varnish
    - include: _lb-external.yml tags=lb-external
    
Doing tags this way allows running of environment.yml playbook with only the server tags you specify.  Saves lots of time in testing.  Could speed up dynamic expansion / contraction if we keep track of which tags need to be run when, say, adding a single fpm node.

The other difference here is the underscore in the playbook names.  I wanted a way to differentiate the "main" playbooks from the "child" playbooks, and the best way I came up with is for the the main ones  (environment,yml, deploy,yml, etc) to be non-underscore-prefixed, and the child ones that generally shouldn't be run directly be prefixed with underscore.  I'm not married to this idea, but not having a way to visually see what the "main" playbooks were was really bothering me.

Preemptive questions and answers:
q) Why not separate the children playbooks in their own dir.
a) Doesn't work, it'll look for tasks in playbooks/child-dir/tasks/ instead of playbooks/tasks/

q) Why shouldn't "child" playbooks be run directly?
a) Some cross-role variables (basically the dynamic backnet addr vars) won't get setup without going through the "main" playbooks.  Those cross-roll variables are tagged with ALL the tags, so they get pulled in when using the --tags version, and it works.

### CODE REVIEW

File: playbooks/tasks/asset/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+
+- name: Extract asset archive
+  unarchive:
+    src="{{ nex_app_url_asset_archive_file }}"
+    dest="{{ nex_app_root }}/{{ nex_app_asset_archive_base }}"
+    remote_src=true
+  become: yes
+  become_user: "{{ nex_app_unix_user }}"
```
Comment (clwells) [inline playbooks/tasks/asset/deploy.yml]: not sure if this is right time to ask this but should this be "when: archive file exists"?

File: playbooks/tasks/db/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+---
+- name: Setup a temp dir for the sql file
+  tempfile:
+    state="directory"
+    suffix="-sql"
+  register: sql_tempdir
+
+- name: Download sql
```
Comment (clwells) [inline playbooks/tasks/db/deploy.yml]: How do we know when to do this?  Or is it always done?

File: playbooks/tasks/asset/deploy.yml (hunk 1 above)
Comment (nexpoehler) [inline playbooks/tasks/asset/deploy.yml]: The source in his case is a URL.  If it doesn't exist or isn't an archive, I think "failure" is the right thing.

File: playbooks/tasks/db/deploy.yml (hunk 2 above)
Comment (clwells) [inline playbooks/tasks/db/deploy.yml]: I read your stark tech issue. We know b/c we use this playbook specifically :)

**Pull request merged**